glib/src/subclass/types.rs

```diff
@@ -247,6 +247,8 @@ pub trait ObjectSubclass: Sized + 'static {
     /// Optional.
     const ABSTRACT: bool = false;
 
+    type Type: ObjectType + crate::object::ObjectSubclassIs<Subclass=Self>;
```

Comment:
Could also update `get_instance()` below now, and maybe other APIs. Every usage of `ObjectSubclass::ParentType` apart from the type registration should be `Type` instead.